scripts/snapshot.ts

```diff
@@ -0,0 +1,163 @@
+import Contracts from '../components/Contracts';
+import { getBalance } from '../test/helpers/Utils';
+import { DeployedContracts, isTenderlyFork } from '../utils/Deploy';
+import { NATIVE_TOKEN_ADDRESS } from '../utils/TokenData';
+import '@nomiclabs/hardhat-ethers';
+import '@typechain/hardhat';
+import Decimal from 'decimal.js';
+import { BigNumber } from 'ethers';
+import fs from 'fs';
+import { ethers, getNamedAccounts } from 'hardhat';
+import 'hardhat-deploy';
+
+interface PoolStats {
+    symbol: string;
+    bntTradingLiquidity: BigNumber;
+    tknTradingLiquidity: BigNumber;
+    stakedBalance: BigNumber;
+    depositLimit: BigNumber;
+    tradingFeePPM: number;
+    depositingEnabled: boolean;
+    tradingEnabled: boolean;
+    spotRate: Decimal;
+    emaRate: Decimal;
+    emaBlockNumber: number;
+    emaCompressedNumerator: BigNumber;
+    emaCompressedDenominator: BigNumber;
+    masterVaultTknBalance: BigNumber;
+    externalProtectionVaultTknBalance: BigNumber;
+    bntFundingLimit: BigNumber;
+    bntRemainingFunding: BigNumber;
+    bntFundingAmount: BigNumber;
+    bnTkn: string;
+    bnTknTotalSupply: BigNumber;
+}
+
+interface GlobalStats {
+    blockNumber: number;
+    masterVaultBntBalance: BigNumber;
+    bntPoolBnBntBalance: BigNumber;
+    bntPoolStakedAmount: BigNumber;
+    vortexBntBalance: BigNumber;
+    networkFeePPM: number;
+    minLiquidityForTrading: BigNumber;
+    lockDuration: number;
+    bnBntTotalSupply: BigNumber;
+}
+
+const gcd = (x: BigNumber, y: BigNumber) => {
+    while (y.gt(0)) {
+        const t = y;
+        y = x.mod(y);
+        x = t;
+    }
+
+    return x;
+};
```

Comment:
As far as i can tell, that does not handle correctly cases where `x < y`.
You can amend it via `[x, y] = x.gt(y) ? [x, y] : [y, x]`.

Reply:
I think that if `x < y`, then the first loop will already switch between them as well.

Reply:
That's right, I didn't stop to think about it, sorry :(